x-pack/plugins/apm/server/routes/suggestions/route.ts

```diff
@@ -30,16 +38,61 @@ const suggestionsRoute = createApmServerRoute({
     const size = await context.core.uiSettings.client.get<number>(
       maxSuggestions
     );
+
     const suggestions = await getSuggestions({
       field,
       searchAggregatedTransactions,
       setup,
       size,
       string,
+      start,
+      end,
+    });
+
+    return suggestions;
+  },
+});
+
+const suggestionsWithTermsRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/suggestions_with_terms',
+  params: t.type({
+    query: t.intersection([
+      t.type({
+        field: t.string,
+        string: t.string,
+        termField: t.string,
+        termValue: t.string,
+      }),
+      rangeRt,
```

Comment:
I'm not sure about this very generic endpoint. But if we are going to do it I think it should take a `filter` instead:

```suggestion
  endpoint: 'GET /internal/apm/suggestions_with_filter',
  params: t.type({
    query: t.intersection([
      t.type({
        field: t.string,
        string: t.string,
        filter: t.string
      })
```

Reply:
And to stay consistent with my other comments, I also think we should rename `field` -> `fieldName` and `string` -> `fieldValue`

```suggestion
  endpoint: 'GET /internal/apm/suggestions_with_filter',
  params: t.type({
    query: t.intersection([
      t.type({
        fieldName: t.string,
        fieldValue: t.string,
        filter: t.string
      })
```

Reply:
@sqren 

> I'm not sure about this very generic endpoint. But if we are going to do it I think it should take a `filter` instead:

I had my doubts about the generic endpoint. IMO it was too implicit with no obvious benefits and currently the only place where I'm using it is in the service environment filter. 

so instead I created the [getSuggestionsByServiceName](https://github.com/elastic/kibana/commit/4eae34b4ad02be647d0e51c565a706104c85200c#diff-741fb8e4f5601e16766676f59428e73622c49f33ab1b58e8bfbc53c823225617R12). it should be more clear what it does

Reply:
https://github.com/elastic/kibana/pull/126679/commits/4eae34b4ad02be647d0e51c565a706104c85200c

Reply:
If we only use it for environments, and since you've already hardcoded `service.name` as the only filter, I'm inclined to say that it should be renamed from `getSuggestionsByServiceName` to `getEnvironmentSuggestions`

Reply:
(and then `fieldName` should be replaced with `service.environment` hardcoded)